src/agentscope/utils/monitor.py

```diff
@@ -244,16 +244,17 @@ def __init__(
 
 
 @contextmanager
-def sqlite_transaction(db_path: str) -> Generator:
+def sqlite_transaction(db_path: str, timeout: float = 30.0) -> Generator:
```

Comment:
Why we need a timeout argument here?

Reply:
the default timeout is only 5 seconds, and when the number of agents is large, it may cause timeout errors, e.g.`database is locked.`

Reply:
So this timeout is actually how long to wait for the db lock?

Reply:
Yes